manifests/templates/block-device.yaml.in

```diff
@@ -0,0 +1,38 @@
+apiVersion: extensions/v1beta1
+kind: Deployment
```

Comment:
Shouldn't this be a DaemonSet so it runs on each node?

Reply:
done

Reply:
This is still a deployment and not a daemonset, and it is probably the reason your functional test is failing intermittently with unable to schedule. See https://kubernetes.io/docs/concepts/workloads/controllers/daemonset/ for an example of a daemonset.

Reply:
it's not failing anymore. I saw you merged stuff. I will rebase it again.

Reply:
No need to rebase, what I merged doesn't conflict.

Reply:
But I would still make this a daemonset, that way you can simplify the functional test, because you don't have to look up the node that has the block device, because all nodes will have one.

Reply:
I will do it in the other PR. I want/need to merge this one ASAP and I do not think this is crucial for now.